store/parameter_store.go

```diff
@@ -292,7 +295,11 @@ func (s *ParameterStore) List(env Environment, service string) ([]SecretIdentifi
 				return []SecretIdentifier{}, err
 			}
 			for _, result := range resp.Parameters {
-				ident, _ := getSecretIDFromParamName(*result.Name)
+				ident, err := getSecretIDFromParamName(*result.Name)
+				if _, ok := err.(*CurrentDeployError); ok {
+					// if any secrets cannot be converted due to CurrentDeployError, skip
+					continue
+				}
```

Comment:
nit: This comment relates to the processing a list of parameters, which means this error should be per parameter it processes. We can be more explicit about the intent.

```suggestion
				// secrets that fail with CurrentDeployError are intended to be read by machines, and not returned for human consumption.
				if _, ok := err.(*CurrentDeployError); ok {
					continue
				}
```